CI/macos/appcast_convert.py

```diff
@@ -0,0 +1,83 @@
+import os
+
+import xmltodict
+
+
+DELTA_BASE_URL = 'https://cdn-fastly.obsproject.com/downloads/sparkle_deltas'
+
+
+def convert_appcast(filename):
+    print('Converting', filename)
+    in_path = os.path.join('output/appcasts', filename)
+    out_path = os.path.join('output/appcasts', filename.replace('_v2', ''))
+    with open(in_path, 'rb') as f:
+        xml_data = f.read()
+    if not xml_data:
+        return
+
+    appcast = xmltodict.parse(xml_data, force_list=('item',))
+
+    new_list = []
+    # Remove anything but the first stable channel item.
+    # Mostly due to some issues with older Sparkle versions
+    # we still have out there.
+    for item in appcast['rss']['channel']['item']:
+        branch = item.pop('sparkle:channel', 'stable')
+        if branch != 'stable':
+            continue
+        # remove delta information (incompatible with Sparkle 1.x)
+        item.pop('sparkle:deltas', None)
+        new_list.append(item)
+        break
+
+    appcast['rss']['channel']['item'] = new_list
```

Comment:
I'd prefer if this function were "pure" and not modify the appcast data structure at all. Operate on a copy, then pass the copy to `xmltodict` or use functional patterns to begin with.